Rewrite frontend styling with spectre.css, less Js, more usage of Django Forms

Akasch opened the pull request:
This rewrites the frontend styling from bootstrap to spectre and refactors parts of the backend. This reduces code in the styling and custom backend code.

Akasch committed:
remove mostly unused moustrap.js

Akasch committed:
remove unused select2

Akasch committed:
convert to spectre, remove JS code for index, footer and aboaut

Akasch committed:
Merge branch 'master' into spectre-css

Akasch committed:
add label texts directly to the model

Akasch committed:
Date/DateTime Poll: show year if other than current

Akasch committed:
Create date/datetime polls: fail early on wrong values, optimize errorcases

Akasch committed:
date and datetimepoll move creation to spectre, reduce JS

Akasch committed:
remove unused script/css

Akasch committed:
vote, poll overview, create generic poll: move to spectre, vix various style issues

Akasch committed:
add django messages to base template

Akasch committed:
settings page to spectre

Akasch committed:
registration to spectre-css

Akasch committed:
refactor navbar

Akasch committed:
index: mobile layout, backgroud color for selected option + cleanup

Akasch committed:
layout: create dt, settings for mobile; fix bug in settings

Akasch committed:
use django password reset form

Akasch committed:
vote: use form, move parts of validation logic to form, add hints to forms, migrations

Akasch committed:
fix poppovers on poll page

Akasch committed:
copy_poll: rework to form, spectrecss

Akasch committed:
front page styling

Akasch committed:
invitation: spectre

Akasch committed:
edit choisvalues: use form + spectre

Akasch committed:
remove static gravatar image from commentedit

Akasch committed:
poll.urls.py: migrate from url() to path()

Akasch committed:
Merge branch 'master' into spectre-css

Akasch committed:
fix datetime edit

wichmannpas commented on bitpoll/base/templates/base.html:
```diff
@@ -27,52 +27,28 @@
+            <div class="container">
+                {% block messages %}
+                    {% for message in messages %}
+                        <div{% if message.tags %} class="alert alert-{{ message.tags }}"{% endif %}>{{ message }}</div>
```
Again, `toast`?

Akasch replied:
the alert class should be gone now

wichmannpas commented on bitpoll/settings.py:
```diff
@@ -128,11 +130,14 @@
-                'css/bootstrap.css',
-                'css/bootstrap-theme.css',
+                #'css/bootstrap.css',
+                #'css/bootstrap-theme.css',
```
Why not remove these lines properly?

wichmannpas commented on bitpoll/settings.py:
```diff
@@ -179,15 +183,12 @@
                 'js/util.js',
-                'js/custom-select.jquery.js',
-                'js/common.js',
+                #'js/custom-select.jquery.js',
```
Why not remove this line properly?

Akasch committed:
fix navbar css, remove alert css, delete unused files'

Akasch committed:
delete unused files

Akasch committed:
spectre: aler->toast, formating, delete commented lines

Akasch committed:
groups code to spectre and forms

Akasch committed:
move settings to spectre, better errorhandling in caldav code

this moves widget_tweaks_extra to django_spectre_css

Akasch committed:
migrate group search from select2 to cusom script

Akasch committed:
move Javascript vorm JQerry to VanillaJS

Akasch committed:
fixes in new javascript code

Akasch committed:
fix spam challenge output on comment form

Akasch committed:
add polyfill for IE 10+

Akasch committed:
due_date: use multifield, generally use render_form_haeder

Akasch committed:
fix styling, fix formating, fix copy poll

Akasch committed:
fix general styling and javascript on universal choice creation

Akasch committed:
allow the ordering of choices in a universal poll withut JS, fix transactional locking in changing choices

Akasch committed:
base.html: fix timezone display, add lang attribute to html tag

Akasch committed:
make timezone warning the same on poll overview and vote

Akasch committed:
update dependencies

Akasch committed:
style fixes, fix UX if no Date/Tie was selected in JS create view

Akasch committed:
fix style and function of poll ownership in settings

Akasch committed:
Merge branch 'master' into spectre-css

Akasch committed:
Merge remote-tracking branch 'github/master' into spectre-css

Akasch committed:
Merge branch 'master' into spectre-css

Akasch committed:
Fix Migration after Markdown

Akasch committed:
Merge branch 'master' into spectre-css

Akasch committed:
add stiky autor collumn.

On desktop monitors make the author collumn sticky so it is always on the left of the frame wihle scrolling. Dos not work(yet) for the vote header as the poll results themselves are not in a scrollabel area

Akasch committed:
Merge branch 'master' into spectre-css

Akasch committed:
Add pyproject.toml; fix icon errors

Akasch committed:
Styling fixes
